scripts/characters/enemies/mouse_trap.gd

```diff
@@ -15,16 +15,17 @@ func _ready():
 # Function called when a body enters the trap
 func _on_body_entered(body: Node) -> void:
```

Comment:
The function _on_body_entered() feels unnecessarily nested. I suggest adding an early return like this:

```gdscript
# Function called when a body enters the trap
func _on_body_entered(body: Node) -> void:
	# Early return if any of these conditions are true:
	# - The trap has already been activated
	# - The entering body is flying
	# - The entering body doesn't have a "die" method
	if has_activated or body.is_flying or not body.has_method("die"):
		return
	
	body.animated_sprite_2d.visible = false
	animated_sprite_2d_2.play("pow")
	
	if body == player:
		animated_sprite_2d_1.play("player_trapped")
	else:
		animated_sprite_2d_1.play("bug_trapped")
	
	body.die()
	has_activated = true
```